### ISSUE TITLE
Mecury parser failed to load GB2312 and GBK webpages

### ISSUE THREAD

**Issue by atbest** [yang991178/fluent-reader#63]
When the encoding of webpage is GBK or GB2312, Mecury parser is unable to read and render the content correctly.

**Comment by atbest** [comment]
I have saw your PR to fix the encoding problem. However, some sites don't provide encoding information in `content-type` of response header, and encoding is usually placed in the html head/meta tags. So the `decodeFetchResponse` function needs to check the html as well. The following code should work for it.

```typescript
const CHARSET_RE = /charset=([^()<>@,;:/[\]?.=\s]*)/i
const CHARSET_META_RE = /<meta .*?charset\s*=['"\s]*([^'"\s]*).*?>/i
const CHARSET_CN_RE = /gbk|gb-?2312/i
export async function decodeFetchResponse(response: Response) {
    const utf8 = "utf-8"
    const gb = "gb18030"
    const buffer = await response.arrayBuffer()
    const ctype = response.headers.has("content-type") && response.headers.get("content-type")
    let charset = (ctype && CHARSET_RE.test(ctype)) ? CHARSET_RE.exec(ctype)[1] : ""
    if (CHARSET_CN_RE.test(charset)) charset = gb
    let decoder = new TextDecoder(charset !== "" ? charset : utf8)
    let html = decoder.decode(buffer)
    if (charset === "") {
        let result = CHARSET_META_RE.exec(html)
        charset = result ? result[1] : utf8
        if (charset !== utf8) {
            if (CHARSET_CN_RE.test(charset)) charset = gb
            decoder = new TextDecoder(charset)
            html = decoder.decode(buffer)
        }
    }
    return html
}
```

**Comment by yang991178** [comment]
Fixed in 5c6cbf5be09bb6b79f86569f6d00374787d0bb0b

**Comment by atbest** [comment]
Great! Thanks.
